test-framework/junit5/src/main/java/io/quarkus/test/junit/TestProfile.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.test.junit;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a 'test profile'. Tests run under a test profile
+ * will have different configuration options to other tests.
+ *
+ * Due to the global nature of Quarkus if a previous test was
+ * run under a different profile then Quarkus will need to be
+ * restarted when the profile changes. Unfortunately there
+ * is currently no way to order tests based on profile, however
+ * this can be done manually by running tests in alphabetical
+ * order and putting all tests with the same profile in the same
+ * package.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface TestProfile {
```

Comment:
IMHO it would make sense to define this interface as `@Inheritable` in case someone wants to define it on a common superclass.

For JUnit 5 tests I am also used to be able to define my own [_composed annotations with meta-annotations_](https://junit.org/junit5/docs/current/user-guide/#writing-tests-meta-annotations), which I think could also be very useful for Quarkus' test annotations like `@QuarkusTestResource` and `@TestProfile`. But for this I will file a separate enhancement request.

Reply:
Please do, that sounds useful

Reply:
See #10297.

Reply:
Should I file a separate enhancement request for declaring `@QuarkusTestResource` and `@TestProfile` as `@Inherited`?

Reply:
Sure yeah

Reply:
> See #10297.

Thannks